runtime/interp.c

```diff
@@ -687,26 +689,31 @@ value caml_interprete(code_t prog, asize_t prog_size)
       *--sp = accu;
       /* Fallthrough */
     Instruct(GETGLOBAL):
-      accu = Field(caml_read_root(caml_global_data), *pc);
+      accu = Field(caml_global_data, *pc);
       pc++;
       Next;
 
     Instruct(PUSHGETGLOBALFIELD):
       *--sp = accu;
       /* Fallthrough */
     Instruct(GETGLOBALFIELD): {
-      accu = Field(caml_read_root(caml_global_data), *pc);
+      accu = Field(caml_global_data, *pc);
       pc++;
       accu = Field(accu, *pc);
       pc++;
       Next;
     }
 
-    Instruct(SETGLOBAL):
-      caml_modify(&Field(caml_read_root(caml_global_data), *pc), accu);
+    Instruct(SETGLOBAL):  {
+      /* temporary variable used in caml_modify,
+         caml_modify cannot be called directly on generational roots */
+      value global_data = caml_global_data;
```

Comment:
I'm a bit confused why we need to do the juggling with `global_data` and `caml_modify_generational_global_root`.
Surely we can alter the field of any OCaml object without worrying if there is a reference registered as a generational global root from another place?

That is, can't we reduce this block to:
```C
caml_modify(&Field(caml_global_data, *pc), accu);
```